docs/user/setup.rst

```diff
@@ -26,7 +26,7 @@ Installing clang and runtime dependencies
 -----------------------------------------
 
 Scala Native requires Clang, which is part of the LLVM toolchain. The
-recommended LLVM version is 3.7 or newer, however, the Scala Native sbt
+recommended LLVM version is 9 or newer, however, the Scala Native sbt
```

Comment:
I believe that the clang installation instructions on approximately line #63 needs to
change from simple `clang` to `clang-9`, possibly with a comment to 
consider using update-alternatives to make it the default clang.

Reply:
With one small comment above, this PR LGTM!  
Thank you for making things current and easier to work with.

I do not want to hold up this PR. Let me float an approach to setup.rst which
I have seen work well in other projects. I wish I could remember which ones.
If devos like the approach, I can submit a follow-on PR, so this one does
not get held up by bike-shedding.

The idea is to deal with base requirements, supported variants, variants
which may work, and those known not to work using the concepts from
statistics of central tendency and variation.  That is, describe the
requirements at a high level (i.e. LLVM not llvm-9), then give
a section which says something like "Scala Native is developed on a 
number of hardware and software configurations. Continuous Integration (CI)
testing is done on the following "known good" configuration:
```
Hardware: 
Operating System: Ubuntu 18.04 LTS
Software:  openjdk-8
                  sbt-org sbt
                  clang-9,  non-LTO, LTO
                   ld.bfd
                  immix GC, boehm GC
                   foo # to test optional component
```
<insert text saying that mac-os (versions???) is not tested but is used by several developers here.>

The Java-8 discussion is a bit tricky.  Saying that scalanative-plugin only supports/works_with  Java-8
  seems to be erecting a strong barrier but, I believe, true.  Is there some softer but also true language
that could be used here?

Reply:
Thanks for the comments @LeeTibbert.

> I believe that the clang installation instructions on approximately line #63 needs to
> change from simple `clang` to `clang-9`, possibly with a comment to
> consider using update-alternatives to make it the default clang.

I intentionally left this out, because the way clang discovery is done at the moment, it does not matter which clang is set the default via alternatives. The build will just pick the latest one.

I agree that the rest of the comments could go in a separate PR.